src/sqliteformat.cpp

```diff
@@ -560,72 +560,84 @@ bool SqliteFormat::modifyComponents(const Incidence::Ptr &incidence, const QStri
             SL3_bind_text(stmt1, index, "", 0, SQLITE_STATIC);
         }
 
-        if (dbop == DBInsert && incidence->created().isNull())
-            incidence->setCreated(QDateTime::currentDateTimeUtc());
-        secs = toOriginTime(incidence->created());
+        if (incidence.created().isValid() || dbop != DBInsert) {
+            secs = toOriginTime(incidence.created());
+        } else {
+            secs = toOriginTime(QDateTime::currentDateTimeUtc());
```

Comment:
The "created" is a public field and used to allow custom values. This now forces the inserted incidences to have current datetime?

Reply:
Maybe the `if ()` condition should be inversed. Actually, this is supposed to do the same thing than the code it is replacing, but it's not touching the incidence.

Previously, in case of DBInsert _without_ created date time, the current date time was saved _and_ set as the created date time of the incidence. Now, it's doing the same, but it's _not_ saving it to the incidence since it's `const`.

Reply:
Suppose you're referring to the setLastModified() part of the commit which got dropped. But this is the created time for which there are no other changes?

For the other thing, yea guess it's slightly better if the incidence is not touched but then again if the next loading of data has an overridden value anyway...

Reply:
Well, no, I'm referring to these removed lines:
```
@@ -560,15 +560,26 @@ bool SqliteFormat::modifyComponents(const Incidence::Ptr &incidence, const QStri
             SL3_bind_text(stmt1, index, "", 0, SQLITE_STATIC);
         }
 
-        if (dbop == DBInsert && incidence->created().isNull())
-            incidence->setCreated(QDateTime::currentDateTimeUtc());
-        secs = toOriginTime(incidence->created());
```
The new `if` is just a negation of the previous one, to use the created() time if possible, or fallback to the current time, but without setting it.

I agree that it's annoying that a save() routine is actually changing the values before saving. Ideally it should not. I keep writing the changed value to the DB to be backward compatible with what was done before.

Actually, modified time and created time are actively used by the sync routines (see ExtendedStorage::insertedIncidences() for instance). For the modified time stamp, I'm quite confident that any modification will lead to the update of it via MemoryCalendar::incidenceUpdated() where there is a call to `setLastModified(QDateTime::currentDateTimeUtc());`. For the created field, it should be set to current time automatically on incidence creation, as seen in Incidence::Incidence() constructor with the call to recreate(). So all in all, these fields should not be empty except when the end user set them so. I would be happy to remove these fallback from mKCal save() routine, but I'm a bit reluctant, not to create unexpected issues.

Reply:
Ah sorry, somehow missed the setCreated() in the old code above. Indeed bad modifying the incidence all the way here.

Debatable whether the created should be forced a valid time at all here, but at least this should work as it used to and quite possible this doesn't really matter too much in practice.